sentry/src/main/java/io/sentry/cache/EnvelopeCache.java

```diff
@@ -202,7 +204,42 @@ public void store(final @NotNull SentryEnvelope envelope, final @NotNull Hint hi
     // write file to the disk when its about to crash so crashedLastRun can be marked on restart
     if (HintUtils.hasType(hint, DiskFlushNotification.class)) {
       writeCrashMarkerFile();
+
+      long timeSinceSdkInit = System.currentTimeMillis() - options.getSdkInitTime();
+      if (timeSinceSdkInit <= options.getStartupCrashDurationThresholdMillis()) {
+        options
+          .getLogger()
+          .log(DEBUG,
+            "Startup Crash detected %d milliseconds after SDK init. Writing a startup crash marker file to disk.",
+            timeSinceSdkInit);
+        writeStartupCrashMarkerFile();
+      }
+    }
+  }
+
+  private void writeStartupCrashMarkerFile() {
+    final File crashMarkerFile = new File(options.getCacheDirPath(), STARTUP_CRASH_MARKER_FILE);
+    try {
+      crashMarkerFile.createNewFile();
+    } catch (Throwable e) {
+      options.getLogger().log(ERROR, "Error writing the startup crash marker file to the disk", e);
+    }
+  }
+
+  public static boolean hasStartupCrashMarker(final @NotNull String dirPath, final @NotNull SentryOptions options) {
+    final File crashMarkerFile = new File(dirPath, STARTUP_CRASH_MARKER_FILE);
+    final boolean exists = crashMarkerFile.exists();
+    if (exists) {
+      if (!crashMarkerFile.delete()) {
```

Comment:
Deleting this file here would return false when doing the same thing for the other outbox integration (because the marker does not exist anymore), I guess you'd need to control the deletion afterward after this one and the other integration gets blocked and flushed.
Also, there's the challenge of the timeout, since it's 2 integrations but a single Future.

Reply:
hmm, but the outbox integration uses a different path, so it should be good? We also check for file existence the line above, before calling `delete`, so it's also not a problem I guess.

I didn't get the single Future part, mind to explain?

Reply:
If each folder has its own marker, that is fine but it's 2 IOs for each file (read/delete) for the same feature, wondering if we can improve this by having only one marker, but still, we'd wait for the execution of both integrations.
Ignore the `Future` part for now.